google/oauth2/service_account.py

```diff
@@ -335,6 +373,12 @@ def refresh(self, request):
         self.token = access_token
         self.expiry = expiry
 
+    @_helpers.copy_docstring(credentials.Credentials)
+    def apply(self, headers, token=None):
+        super(Credentials, self).apply(headers, token=token)
```

Comment:
Shouldn't this come after the headers manipulation?

Reply:
I think either order works - the `headers` object is passed into the function and modified.

https://github.com/googleapis/google-auth-library-python/blob/37141e4dffc2ba3f3f57c5914544fb8b9cf7d017/google/auth/credentials.py#L94-L104